formidling/src/main/java/no/nav/ung/sak/formidling/vedtak/VedtaksbrevGenerererTjenesteUng.java

```diff
@@ -17,43 +22,61 @@
 import no.nav.ung.sak.formidling.template.dto.TemplateDto;
 import no.nav.ung.sak.formidling.template.dto.felles.FellesDto;
 import no.nav.ung.sak.formidling.template.dto.felles.MottakerDto;
+import no.nav.ung.sak.formidling.vedtak.regler.BehandlingVedtaksbrevResultat;
 import no.nav.ung.sak.formidling.vedtak.regler.Vedtaksbrev;
+import no.nav.ung.sak.formidling.vedtak.regler.VedtaksbrevReglerUng;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.List;
+
 @ApplicationScoped
-public class VedtaksbrevGenerererTjenesteImpl implements VedtaksbrevGenerererTjeneste {
+@BehandlingTypeRef(BehandlingType.FØRSTEGANGSSØKNAD)
+@BehandlingTypeRef(BehandlingType.REVURDERING)
+@FagsakYtelseTypeRef
+public class VedtaksbrevGenerererTjenesteUng implements VedtaksbrevGenerererTjeneste {
 
-    private static final Logger LOG = LoggerFactory.getLogger(VedtaksbrevGenerererTjenesteImpl.class);
+    private static final Logger LOG = LoggerFactory.getLogger(VedtaksbrevGenerererTjenesteUng.class);
 
     private BehandlingRepository behandlingRepository;
     private PdfGenKlient pdfGen;
     private ManueltVedtaksbrevInnholdBygger manueltVedtaksbrevInnholdBygger;
     private BrevMottakerTjeneste brevMottakerTjeneste;
+    private VedtaksbrevReglerUng vedtaksbrevRegler;
+
+    public VedtaksbrevGenerererTjenesteUng() {
+    }
 
     @Inject
-    public VedtaksbrevGenerererTjenesteImpl(
+    public VedtaksbrevGenerererTjenesteUng(
         BehandlingRepository behandlingRepository,
         PdfGenKlient pdfGen,
-        ManueltVedtaksbrevInnholdBygger manueltVedtaksbrevInnholdBygger, BrevMottakerTjeneste brevMottakerTjeneste) {
-
+        ManueltVedtaksbrevInnholdBygger manueltVedtaksbrevInnholdBygger, BrevMottakerTjeneste brevMottakerTjeneste,
+        @Any VedtaksbrevReglerUng vedtaksbrevRegler) {
         this.behandlingRepository = behandlingRepository;
         this.pdfGen = pdfGen;
         this.manueltVedtaksbrevInnholdBygger = manueltVedtaksbrevInnholdBygger;
         this.brevMottakerTjeneste = brevMottakerTjeneste;
+        this.vedtaksbrevRegler = vedtaksbrevRegler;
     }
 
-    public VedtaksbrevGenerererTjenesteImpl() {
-    }
 
-    @WithSpan
     @Override
-    public GenerertBrev genererAutomatiskVedtaksbrev(VedtaksbrevGenerererInput vedtaksbrevGenerererInput) {
-        return BrevGenereringSemafor.begrensetParallellitet(() -> doGenererAutomatiskVedtaksbrev(vedtaksbrevGenerererInput));
+    public GenerertBrev genererAutomatiskVedtaksbrev(Behandling behandling, DokumentMalType dokumentMalType, boolean kunHtml) {
```

Comment:
jeg ser at de stedene dette kalles så kalles vedtaksbrevRegler.kjør i forkant uansett, bortsett fra VedtaksbrevBestillingTask. Er det ikke like greit å kalle kjør i forkant så slipper man ekstra kall? Jeg kan godt gjøre det. 

Også går da an å merge VedtaksbrevGenerer tjenestene til ung og klage - må bare sende som  DetaljertResultat. 

Ser det noe i veien for det?